### ISSUE TITLE
PLE0604: false positive?

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#26817]
### Summary

```
  __all__ = ['1']
E __all__ += '2' # PLE0605: Invalid format for `__all__`, must be `tuple` or `list`
```

### Version

0.15.21

**Comment by ntBre** [comment]
It looks like a subset of this was fixed in https://github.com/astral-sh/ruff/pull/4967, but maybe we should push it further.

**Comment by Andrej730** [comment]
I'm not sure it's a bug. Otherwise `__all__ = "Foo"` could also be considered to be a bug, since `"Foo"` is also a sequence (of one-character strings), so `__all__` sort of acts like `["F", "o", "o"]`, while user is likely intended `["Foo"]`.

So allowing `+= "bar"` allows to silently add `["b", "a", "r"]` to the list, making it error-prone. `+= string` makes sense only if `string` is one character long, which is very rare.

Maybe diagnostic message can be clarified since it's not just about `__all__` itself being a list or a tuple, but also covering case when `__all__` is filled with unpacked strings by accident.

**Comment by ntBre** [comment]
Oh you're right! I forgot that `+=` adds the characters of the string to the list:

```pycon
>>> __all__ = ['1']
>>> __all__ += "Foo"
>>> __all__
['1', 'F', 'o', 'o']
```

I was picturing the result as `['1', 'Foo']`. This is already allowed by the rule and produces my expected result:

```py
__all__ = ['1']
__all__ += ['Foo']
```

This does seem like more of a diagnostic or documentation issue to me. @spaceone did you run into a situation where you really intended to add the individual characters of a string to `__all__`?

**Comment by spaceone** [comment]
no, I did not intend that.
I originally had:
`__all__.append('Foo')` but that was flagged as `PYI056: Calling `.append()` on `__all__` may not be supported by all type checkers (use `+=` instead)`

**Comment by spaceone** [comment]
Well, I would like to re-use this issue for a follow up problem instead of a new one: `PLE0604`

```python
__all__ = []


class Foo: pass

K = {'Foo': Foo}

for member in K.values():
    name: str = member.__name__
    __all__ += [name]  # PLE0604: Invalid object in `__all__`, must contain only strings
```

**Comment by ntBre** [comment]
This new problem was previously closed as not-planned in https://github.com/astral-sh/ruff/issues/10924, but there was also a note there about updating the docs, so the https://github.com/astral-sh/ruff/labels/documentation label still seems appropriate.
